plugins/flashrom/fu-flashrom-lspcon-i2c-spi-device.c

```diff
@@ -101,14 +123,63 @@ fu_flashrom_lspcon_i2c_spi_device_open (FuDevice *device,
 	FuDeviceClass *klass =
 		FU_DEVICE_CLASS (fu_flashrom_lspcon_i2c_spi_device_parent_class);
 	g_autofree gchar *temp = NULL;
+	g_autofree gchar *bus_path = NULL;
+	gint bus_fd = -1;
 
 	/* flashrom_programmer_init() mutates the programmer_args string. */
 	temp = g_strdup_printf ("bus=%d", self->bus_number);
 	fu_flashrom_device_set_programmer_args (flashrom_device, temp);
 
+	/* open the bus, not the device represented by self */
+	bus_path = g_strdup_printf("/dev/i2c-%d", self->bus_number);
+	g_debug ("communicating with device on %s", bus_path);
+	if ((bus_fd = g_open (bus_path, O_RDWR, 0)) == -1) {
+		g_set_error (error, G_IO_ERROR, g_io_error_from_errno (errno),
+			     "failed to open %s read-write", bus_path);
+		return FALSE;
+	}
+	fu_udev_device_set_fd (FU_UDEV_DEVICE (self), bus_fd);
+	fu_udev_device_set_flags (FU_UDEV_DEVICE (self), FU_UDEV_DEVICE_FLAG_NONE);
+
 	return klass->open (device, error);
 }
 
+static gint
+probe_active_flash_partition (FuFlashromLspconI2cSpiDevice *self, GError **error)
+{
+	gint bus_fd = fu_udev_device_get_fd (FU_UDEV_DEVICE (self));
+	guint8 data = 0x0e;
+
+	/* device must have been opened */
+	g_return_val_if_fail (bus_fd != -1, -1);
+
+	/* read register 0x0e on page 5, which is set to the currently-running
+	 * flash partition number */
+	if (ioctl (bus_fd, I2C_SLAVE, 0x9a >> 1) == -1) {
```

Comment:
fu_udev_device_ioctl?

Reply:
Kind of awkward casting the bus address to a pointer, but done.